How to log exceptions
=====================

Logging the exception tracks the code, but not the data.  You can walk
through the frames to get the values for the various local values.

The program rgl/fun_with_exceptions.py has a contrived example.  Each
function has a local object x::

    def f():
        x = 'inside f'
        return g()

    def g():
        x = 'inside g'
        return h()

    def h():
        x = 'inside h'
        return i()

    def i():

        x = 'inside i'
        1/0

Here's the output::

    $ python rgl/fun_with_exceptions.py
    DEBUG:root:{'x': 'inside f'}
    DEBUG:root:{'x': 'inside g'}
    DEBUG:root:{'x': 'inside h'}
    DEBUG:root:{'x': 'inside i'}
    DEBUG:root:all done

